src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md

```diff
@@ -0,0 +1,271 @@
+
+- External client
+  - Average Throughput (IOPS)
+  - Average and percentile latency
+- Background Scrub
+  - Average scrub throughput (MiB/s)
+  - Scrub Duration (s)
+- Background Recovery
+  - Average recovery throughput
+  - Number of misplaced objects recovered per second
+
+## Test Environment
+
+A single node with the following configuration was used for the tests:
+
+- Software Configuration: CentOS Stream release 8 Linux Kernel 5.9.1-1.el8.elrepo.x86_64
+- CPU: 2 x Intel(R) Xeon(R) Platinum 8276M CPU @ 2.20GHz
+- nproc: 40
+- System Memory: 64 GiB
+- Tuned-adm Profile: network-latency
+- CephVer: ceph version 17.2.0-176-g6551f450487 (6551f450487afe6c3f7bdd7a7c9c891e447045ed) quincy (stable)
+- Storage: Intel® NVMe SSD DC P4510 Series (SSDPE2KX080T8)
+
+All the Ceph pools were configured with a replication factor of 3. A total of 4 OSDs were configured on the node for the tests.
+
+## Test Methodology
+
+We used the CBT testing framework to run the tests mentioned above. We added 2 new tests to CBT:
+
+1.  To run scrub and client operations in parallel
+2.  To run scrub, recovery and client options in parallel
+
+These tests were first run with osd_op_queue set to WPQ in order to establish a benchmark which we could use to compare the mClock test results against.
+We then ran the test with the osd_op_queue set to mClock scheduler.
+There were 3 repeatability runs carried out with WPQ and mClock each, and we compared the average of these results to avoid any outliers.
+
+## Establish Baseline Client Throughput (IOPS)
+
+For the tests, NVMe SSDs were used as the backing device for the OSDs. A 1:1 mapping of an OSD to the device was set up. But before the actual tests, the baseline throughput was established by running a ‘fio’ benchmark on a RBD pool configured with the same OSDs to be used in the tests and with a replication factor of 3. For this study, an average baseline client throughput of around 15K IOPS (58.6 MiB/s) @ 4KiB random writes was established using the benchmark test. The bluestore throttle parameters, namely [bluestore_throttle_bytes](https://docs.ceph.com/en/latest/rados/configuration/bluestore-config-ref/#confval-bluestore_throttle_bytes) and [bluestore_throttle_deferred_bytes](https://docs.ceph.com/en/latest/rados/configuration/bluestore-config-ref/#confval-bluestore_throttle_deferred_bytes) were determined to be 2 MiB using [these](https://ceph--46700.org.readthedocs.build/en/46700/rados/configuration/mclock-config-ref/#benchmarking-test-steps-using-osd-bench) steps (Note that ‘fio’ tool was used instead of OSD bench).
+
+| Device Type | Number of OSDs |            RBD Pool Configuration            | Baseline Throughput(@4KiB Random Writes, QD: 64) |
+| :---------: | :------------: | :------------------------------------------: | :----------------------------------------------: |
+|  NVMe SSD   |       4        | Replication Factor: 3 / pg_num & pgp_num: 64 |          15007.16965 IOPS (58.6 MiB/s)           |
```

Comment:
The formatting looks a little off here (Device Type is right next to Number of OSD). I think Sridhar fixed this in his post by adding a line break.